# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../lftp/lftp-4.4.0-glibc-2.16-no-gets.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 6c925916dce3505dab1385e14dbed2d7b9f727e4 Mon Sep 17 00:00:00 2001
From: Christian Wiese <chris@opensde.org>
Date: Tue, 23 Oct 2012 10:47:52 +0200
Subject: [PATCH] fixed to build with glibc >= 2.16 where 'gets' is not defined in the gnulib anymore

---
 lib/stdio.in.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/lib/stdio.in.h b/lib/stdio.in.h
index b1b543d..2d62f24 100644
--- a/lib/stdio.in.h
+++ b/lib/stdio.in.h
@@ -702,10 +702,12 @@ _GL_WARN_ON_USE (getline, "getline is unportable - "
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning; besides, C11
    removed it.  */
+#if defined gets
 #undef gets
 #if HAVE_RAW_DECL_GETS
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif // if defined gets
 
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
-- 
1.7.2.3

